platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -130,26 +162,61 @@ class HangingProtocolService {
    * @param {string} name strategy name
    * @param {Function} callback image loader callback
    */
-  registerImageLoadStrategy(name, callback) {
+  public registerImageLoadStrategy(name, callback) {
     if (callback instanceof Function && name) {
       this.registeredImageLoadStrategies[name] = callback;
     }
   }
 
-  setHangingProtocolAppliedForViewport(i) {
+  public setHangingProtocolAppliedForViewport(i) {
     this.hpAlreadyApplied[i] = true;
   }
 
   /**
    * Adds a custom setting that can be chosen in the HangingProtocol UI and applied to a Viewport
-   *
-   * @param settingId The ID used to refer to the setting (e.g. 'displayCADMarkers')
-   * @param settingName The name of the setting to be displayed (e.g. 'Display CAD Markers')
-   * @param options
+   * It is possible to add more than one setting to a given id, in which case all are called.
+   * @param id The ID used to refer to the setting (e.g. 'displayCADMarkers')
+   * @param name The name of the setting to be displayed (e.g. 'Display CAD Markers')
    * @param callback A function to be run after a viewport is rendered with a series
+   * @param options
+   * @param options.store is a function to be called when displaying a new display set, to record any existing values
    */
-  addCustomViewportSetting(...params) {
-    this.customViewportSettings.push(...params);
+  public addCustomViewportSetting(id, name, callback, store, options) {
+    this.customViewportSettings.push({ id, name, store, callback, options });
+  }
+
+  /** Do the inverse store operation to store information between views. */
+  public applyCustomViewportStore(viewportOptions, priorViewportOptions, ...args) {
+    if (!priorViewportOptions) return;
+    for (const setting of this.customViewportSettings) {
+      if (!setting.store) continue;
+      setting.store(viewportOptions, priorViewportOptions, ...args);
+    }
+  }
+
+  /** Runs the specified custom viewport setting, on the provided viewport */
+  public applyCustomViewportSettings(viewportOptions, viewport, ...args) {
+    const { customViewportOptions } = viewportOptions;
+    if (!customViewportOptions) return;
+    Object.keys(customViewportOptions).forEach(key => {
+      this.applySingleCustomViewportSetting(
+        key,
+        customViewportOptions[key],
+        viewport,
+        ...args
+      );
+    })
+  }
+
+  public applySingleCustomViewportSetting(id, value, viewport, ...args) {
+    let callCount = 0;
+    for (const setting of this.customViewportSettings) {
+      // Can have pure storage settings as well.
+      if (setting.id !== id || !setting.callback) continue;
+      setting.callback(id, value, viewport, ...args);
+      callCount += 1;
+    }
+    if (callCount === 0) console.warn("No custom setting found for", id);
   }
```

Comment:
See my general comment for these in the `CornerstoneViewportService`

Reply:
The HangingProtocolService owns the application of the hanging protocol, and the custom viewport settings were previously owned by the HPS, so I would claim they still should be owned.
The store service is part of the HPS because it describes how to apply newly displayed data in the context of an existing hanging protocol.